### PR TITLE
conditionally execute more tests from the github action

### PR DESCRIPTION
**Pull request by lmsurpre**
Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -43,7 +43,16 @@ jobs:
...
+        if [ -n "$(git --no-pager diff --name-only ${BASE} ${GITHUB_SHA} -- fhir-search fhir-persistence fhir-persistence-jdbc)" ]; then
+          echo "Changes affect fhir-persistence-jdbc, running ALL"
+          mvn -B -T2C package --file fhir-parent/pom.xml -P jdbc-all-tests --no-transfer-progress
+        else
+          echo "No changes affect fhir-persistence-jdbc, running MINIMAL"
+          mvn -B -T2C package --file fhir-parent/pom.xml --no-transfer-progress
+        fi
   e2e-tests:
     runs-on: ubuntu-latest
     strategy:
```
Comment (albertwang-ibm) [inline .github/workflows/build.yml]: should we add fhir-persistence-schema also?

File: fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/Main.java [hunk 2]
```diff
@@ -94,9 +94,6 @@
     // The thread pool used for concurrent request processing
     private ExecutorService pool;
 
-    // number of ms to wait for the thread pool to shut down
-    private long poolShutdownWaitTime = 60000;
-
     private List<FileExpectation> fileExpectations = new ArrayList<>();
 
     // mode of operation
```
Comment (albertwang-ibm) [inline fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/Main.java]: does this mean poolShutdownWaitTime has never been used?

File: .github/workflows/build.yml (hunk 1 above)
Comment (lmsurpre) [reply .github/workflows/build.yml]: sure.  included in update

File: fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/Main.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/Main.java]: correct, eclipse warned that this value is never used and so i removed it.

however, to be completely honest, i was just looking for an edit i could make in jdbc to verify the conditional logic.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -43,7 +43,42 @@ jobs:
     - name: Build samples
       run: mvn -B install --file fhir-examples/pom.xml --no-transfer-progress
     - name: Build parent with tests
-      run: mvn -B -T2C package --file fhir-parent/pom.xml -P jdbc-all-tests --no-transfer-progress
+      env:
```
Comment (prb112) [inline .github/workflows/build.yml]: excellent

**Pull request merged**